neqo-transport/src/path.rs

```diff
@@ -168,11 +180,37 @@ impl Paths {
         }
     }
 
-    pub fn lost(&mut self, lost: [u8; 8]) {
+    pub fn lost_challenge(&mut self, lost: [u8; 8]) {
         for p in &self.paths {
             p.borrow_mut().lost(lost);
         }
     }
+
+    pub fn write_frames(
+        &mut self,
+        builder: &mut PacketBuilder,
+        tokens: &mut Vec<RecoveryToken>,
+        stats: &mut FrameStats,
+    ) {
+        while let Some(seqno) = self.to_retire.pop() {
+            if builder.remaining() < 1 + Encoder::varint_len(seqno) {
+                self.to_retire.push(seqno);
+                break;
+            }
+            builder.encode_varint(FRAME_TYPE_RETIRE_CONNECTION_ID);
+            builder.encode_varint(seqno);
+            tokens.push(RecoveryToken::RetireConnectionId(seqno));
+            stats.retire_connection_id += 1;
```

Comment:
Good call.

Reply:
Hang on.  We generally count frames sent, including retransmissions.  Counting when we decide to retire would give us a different count as it could be some time until we put the frame out.  I'd rather get a precise accounting of the frames we send, and maybe look into counting losses so that we can get information on retransmissions - or remove them from counts - in a different way.

Reply:
I think we need another stats that counts retired connectionId not frames.